src/Sylius/Component/Core/Statistics/Provider/SalesStatisticsProvider.php

```diff
@@ -15,46 +15,94 @@
 {
     /** @var array<string, string> */
     private array $formatsMap = [];
 
-    /** @param array<string, array{interval: string, period_format: string}> $intervalsMap */
+    /** @var array<StatisticsProviderRegistryInterface> */
+    private array $statisticsProviderRegistries = [];
+
+    /**
+     * @param array<string, array{interval: string, period_format: string}> $intervalsMap
+     * @param iterable<StatisticsProviderRegistryInterface> $statisticsProviderRegistries
+     */
     public function __construct(
         private OrdersTotalsProviderRegistryInterface $ordersTotalsProviderRegistry,
         array $intervalsMap,
+        iterable|null $statisticsProviderRegistries = null,
     ) {
         foreach ($intervalsMap as $type => $intervalMap) {
             $this->formatsMap[$type] = $intervalMap['period_format'];
         }
+
+        if ($statisticsProviderRegistries === null) {
+            trigger_deprecation(
+                'sylius/core',
+                '2.1',
+                'Not passing $statisticsProviderRegistries through constructor is deprecated and will be prohibited in Sylius 3.0.',
+            );
+
+            return;
+        }
+
+        Assert::allIsInstanceOf(
+            $statisticsProviderRegistries,
+            StatisticsProviderRegistryInterface::class,
+            sprintf('All statistics provider registries should implement the "%s" interface.', StatisticsProviderRegistryInterface::class),
+        );
+        $this->statisticsProviderRegistries = $statisticsProviderRegistries instanceof \Traversable ? iterator_to_array($statisticsProviderRegistries) : $statisticsProviderRegistries;
     }
 
     public function provide(string $intervalType, \DatePeriod $datePeriod, ChannelInterface $channel): array
     {
         $format = $this->getPeriodFormat($intervalType);
 
-        $sales = $this->ordersTotalsProviderRegistry
-            ->getByType($intervalType)
-            ->provideForPeriodInChannel($datePeriod, $channel)
-        ;
+        if ($this->statisticsProviderRegistries == []) {
+            $sales = $this->ordersTotalsProviderRegistry
+                ->getByType($intervalType)
+                ->provideForPeriodInChannel($datePeriod, $channel)
+            ;
 
-        return $this->withFormattedDates($sales, $format);
+            return $this->withFormattedDates($sales, $format);
+        }
+
+        $data = [];
+        foreach ($this->statisticsProviderRegistries as $statisticsProviderRegistry) {
+            $result = $statisticsProviderRegistry
+                ->getByType($intervalType)
+                ->provideForPeriodInChannel($datePeriod, $channel)
+            ;
+
+            if ($data === []) {
+                $data = $result;
+
+                continue;
+            }
+
+            $data = $this->mergeArraysByPeriod($data, $result);
+        }
+
+        return $this->withFormattedDates($data, $format);
```

Comment:
Maybe add some caching mechanism in here? Something like half an hour shouldn't make much of a difference 🤷